client/my-sites/site-settings/podcasting-details/link.jsx

```diff
@@ -113,20 +117,23 @@ class PodcastingLink extends Component {
 export default connect( ( state, ownProps ) => {
 	const { fields } = ownProps;
 
-	const podcastingCategoryId = Number( fields && fields.podcasting_category_id );
-	const isPodcastingEnabled = podcastingCategoryId > 0;
-
 	const siteId = getSelectedSiteId( state );
 	const siteSlug = getSelectedSiteSlug( state );
 
+	const podcastingCategoryId = Number( fields && fields.podcasting_category_id );
```

Comment:
did you mean to flatten a lack of data into a `0`? seems like `fields && Number( field.podcasting_category_id )` would be more normal.

also, why do we have `snake_case` names here? seems like they should have been eliminated when data came into Calypso at the WPCOM API interface

Reply:
I don't have a preference for the first thing. If you want to submit a PR, we will accept it.

For the settings names, if you want to submit a PR to fix this for all settings across Calypso...... Good luck and godspeed. You might want to take a look at data type handling too, which is the reason we need the `Number` stuff at the interface between the settings code and the podcasting components.

Reply:
> If you want to submit a PR, we will accept it.

really wasn't a suggestion (definitely not yet) - more like, "was this what you wanted to happen because that's what's happening"

> For the settings names, if you want to submit a PR to fix this for all settings across Calypso...... Good luck and godspeed. You might want to take a look at data type handling too, which is the reason we need the Number stuff at the interface between the settings code and the podcasting components.

well this is stuff we are trying to root out of Calypso. the interface should be in the data layer in `fromApi` where the data enters the app. are you saying these settings were already existing and here we're simply shuffling around existing settings?

Reply:
> was this what you wanted to happen because that's what's happening

I think so - we wanted to have the podcasting category ID always be a number, and we usually check if podcasting is enabled with `podcastingCategoryId > 0`.  There would be no change in behavior either way though.

> are you saying these settings were already existing and here we're simply shuffling around existing settings?

The code that retrieves and updates settings is shared by all settings screens, which is good, but it has a lot of legacy behavior and other quirks.  You can start investigating this code here:  https://github.com/Automattic/wp-calypso/blob/df6bc312/client/my-sites/site-settings/wrap-settings-form.jsx

> should be in the data layer

Per https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#app-components, query components are still the official recommendation for how we handle data needs.

I am all for implementing a different, and probably technically superior approach, but first, please make sure it is documented as the preferred replacement for what came before it.

Reply:
If they are existing settings that's fine. We can't rewrite the whole app 🤷‍♂️ 

Not sure about the mention of query components. That's quite an independent thing.

Better docs are still coming…pending some in-progress major data discussions.